### PR TITLE
feat: print fancy messages for errors

### PR DESCRIPTION
**Pull request by hyf0**
### Description

Just found that we need `ariadne` for better flexbility to control how we display the messages, which is useful when we support warnings.

### Test Plan

---

### CODE REVIEW

**Comment by hyf0** [comment]
Current dependencies on/for this PR:
* main
  * **PR #315** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/315?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #316** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/316?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #317** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/317?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #319** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/319?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #320** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/320?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #321** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/321?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #322** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/322?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #323** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/323?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                  * **PR #324** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/324?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #325** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/325?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                      * **PR #326** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/326?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                        * **PR #327** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/327?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                          * **PR #328** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/328?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                            * **PR #329** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/329?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
                              * **PR #332** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/332?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/329?utm_source=stack-comment).

File: crates/rolldown/tests/fixtures/errors/unresolved_entry/artifacts.snap [hunk 1]
```diff
@@ -8,5 +8,12 @@ input_file: crates/rolldown/tests/fixtures/errors/unresolved_entry
 ## UNRESOLVED_ENTRY
 
 ```text
-Cannot resolve entry module ./main.js.
+[UNRESOLVED_ENTRY] Error: Unresolved entry module
+   ╭─[Output:1:1]
```
Comment (underfin) [inline crates/rolldown/tests/fixtures/errors/unresolved_entry/artifacts.snap:12]: I think the user maybe will be curious where is `Output:1:1`

Comment (hyf0) [reply crates/rolldown/tests/fixtures/errors/unresolved_entry/artifacts.snap:12]: I feel the same. But I currently don't know how to config `ariadne` to make it disappear. Probably need to create an issue with a `help wanted` label :).

**Review by underfin — APPROVED** [review]

**Comment by hyf0** [comment]
### Merge activity

* **Nov 20, 1:16 AM**: @hyf0 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/329).
* **Nov 20, 1:16 AM**: @hyf0 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/329).

**Pull request merged**